python/ray/tune/trial_runner.py

```diff
@@ -385,6 +421,15 @@ def has_resources(self, resources):
         """Returns whether this runner has at least the specified resources."""
         return self.trial_executor.has_resources(resources)
 
+    def _stop_experiment_if_needed(self):
```

Comment:
as its currently written would this kill _all_ experiment trials for `run_experiments(..., concurrent=True)`?